cli/command/task.go

```diff
@@ -10,6 +14,157 @@ func NewTaskCommand(b *CommandBase) *TaskCommand {
 	return &TaskCommand{b}
 }
 
-func (e *TaskCommand) Command() cli.Command {
-	return cli.Command{}
+func (t *TaskCommand) Command() cli.Command {
+	return cli.Command{
+		Name:  "task",
+		Usage: "manage layer0 tasks",
+		Subcommands: []cli.Command{
+			{
+				Name:      "create",
+				Usage:     "create a new task",
+				Action:    t.create,
+				ArgsUsage: "ENVIRONMENT NAME DEPLOY",
+				Flags: []cli.Flag{
+					cli.IntFlag{
+						Name:  "copies",
+						Value: 1,
+						Usage: "number of copies of deploy to run (default: 1)",
+					},
+					cli.StringSliceFlag{
+						Name:  "env",
+						Usage: "environment variable override in format 'CONTAINER:VAR=VAL' (can be specified multiple times)",
+					},
+				},
+			},
+			{
+				Name:      "delete",
+				Usage:     "delete a Task",
+				ArgsUsage: "NAME",
+				Action:    t.delete,
+			},
+			{
+				Name:      "list",
+				Usage:     "list all Tasks",
+				Action:    t.list,
+				ArgsUsage: " ",
+			},
+			{
+				Name:      "read",
+				Usage:     "describe a Task",
+				Action:    t.read,
+				ArgsUsage: "NAME",
+			},
+			{
+				Name:      "logs",
+				Usage:     "get the logs for a task",
+				Action:    t.logs,
+				ArgsUsage: "NAME",
+				Flags: []cli.Flag{
+					cli.IntFlag{
+						Name:  "tail",
+						Usage: "number of lines from the end to return",
+					},
+					cli.StringFlag{
+						Name:  "start",
+						Usage: "the start of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+					cli.StringFlag{
+						Name:  "end",
+						Usage: "the end of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+				},
+			},
+		},
+	}
+}
+
+func (t *TaskCommand) create(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "ENVIRONMENT", "DEPLOY", "NAME")
+	if err != nil {
+		return err
+	}
+
+	environmentID, err := t.resolveSingleEntityIDHelper("environment", args["ENVIRONMENT"])
+	if err != nil {
+		return err
+	}
+
+	deployID, err := t.resolveSingleEntityIDHelper("deploy", args["DEPLOY"])
+	if err != nil {
+		return err
+	}
+
+	req := models.CreateTaskRequest{
```

Comment:
We need to add back in the container overrides to this model/function

Reply:
I noticed https://github.com/quintilesims/layer0/blob/03b20129285df3e1b92da3393ec2844c405768dd/common/models/container_override.go

this isn't in the cli-refactor is this still needed to do the override.

Reply:
Should I add the parse override to the util.go?

Reply:
I think it makes sense to keep it in this file, since this file will be the only one using it.